doit/dependency.py

```diff
@@ -593,17 +597,23 @@ def get_status(self, task, tasks_dict):
 
         # no dependencies means it is never up to date.
         if not (task.file_dep or uptodate_result_list):
+            if rebuild_log is not None:
+                rebuild_log.append('Task has no dependencies.')
             return 'run'
 
         # if target file is not there, task is not up to date
         for targ in task.targets:
             if not os.path.exists(targ):
                 task.dep_changed = list(task.file_dep)
+                if rebuild_log is not None:
+                    rebuild_log.append('Target file "{}" does not exist.'.format(targ))
```

Comment:
It would be better to not repeat the text of the message all the time. I mean I prefer a message like

Target file does not exist:
 - target1
 - target2
 - ...